File: frontend/src/app/core/services/patient-account-link.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import {
  PatientAccountCandidate,
  PatientAccountLink,
  PatientAccountLinkPatient,
  VerifyPatientAccountLinkRequest
} from '../models/patient-account-link.model';

@Injectable({ providedIn: 'root' })
export class PatientAccountLinkService {
  private readonly apiBaseUrl = environment.apiBaseUrl;

  constructor(private readonly http: HttpClient) {}

  getPatientByNumber(patientNumber: string): Observable<PatientAccountLinkPatient> {
    return this.http.get<PatientAccountLinkPatient>(`${this.apiBaseUrl}/api/patient-account-links/patient`, {
      params: { patientNumber }
    });
  }

  searchPatientAccounts(query: string): Observable<PatientAccountCandidate[]> {
    return this.http.get<PatientAccountCandidate[]>(`${this.apiBaseUrl}/api/patient-account-links/accounts`, {
      params: { query }
    });
  }

  verifyLink(request: VerifyPatientAccountLinkRequest): Observable<PatientAccountLink> {
    return this.http.post<PatientAccountLink>(`${this.apiBaseUrl}/api/patient-account-links/verify`, request);
  }
}

File: backend/src/main/java/com/doctorapp/medicaltracker/service/PatientAccountLinkService.java
package com.doctorapp.medicaltracker.service;

import java.util.List;

import com.doctorapp.medicaltracker.dto.PatientAccountCandidateResponse;
import com.doctorapp.medicaltracker.dto.PatientAccountLinkPatientResponse;
import com.doctorapp.medicaltracker.dto.PatientAccountLinkResponse;
import com.doctorapp.medicaltracker.dto.VerifyPatientAccountLinkRequest;

public interface PatientAccountLinkService {

    PatientAccountLinkPatientResponse getPatientByNumber(String patientNumber);

    List<PatientAccountCandidateResponse> searchPatientAccounts(String query);

    PatientAccountLinkResponse verifyLink(VerifyPatientAccountLinkRequest request);
}
